File: ios/ALCConstants.m
//
//  ALCConstants.m
//  react-native-pure-navigation
//
//  Created by skylar on 2020/9/21.
//

#import "ALCConstants.h"

NSString * const NAVIGATION_EVENT   = @"NAVIGATION_EVENT";
NSString * const EVENT_TYPE         = @"EVENT_TYPE";
NSString * const COMPONENT_RESULT   = @"COMPONENT_RESULT";
NSString * const RECLICK_TAB        = @"RECLICK_TAB";
NSString * const RESULT_TYPE        = @"RESULT_TYPE";
NSString * const RESULT_TYPE_OK     = @"RESULT_TYPE_OK";
NSString * const RESULT_TYPE_CANCEL = @"RESULT_TYPE_CANCEL";
NSString * const RESULT_DATA        = @"RESULT_DATA";
NSString * const SCREEN_ID          = @"SCREEN_ID";

@implementation ALCConstants
RCT_EXPORT_MODULE(ALCConstants);

- (NSDictionary<NSString *, NSString *> *)constantsToExport {
    return @{
             @"NAVIGATION_EVENT": NAVIGATION_EVENT,
             @"EVENT_TYPE": EVENT_TYPE,
             @"RECLICK_TAB": RECLICK_TAB,
             @"COMPONENT_RESULT": COMPONENT_RESULT,
             @"RESULT_TYPE": RESULT_TYPE,
             @"RESULT_TYPE_OK": RESULT_TYPE_OK,
             @"RESULT_TYPE_CANCEL": RESULT_TYPE_CANCEL,
             @"RESULT_DATA": RESULT_DATA,
             @"SCREEN_ID": SCREEN_ID,
            };
}

- (dispatch_queue_t)methodQueue {
    return dispatch_get_main_queue();
}

+ (BOOL)requiresMainQueueSetup {
    return YES;
}

@end

File: android/src/main/java/com/reactnativepurenavigation/ALCConstants.kt
package com.reactnativepurenavigation

import com.facebook.common.logging.FLog
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.ReadableMap

class ALCConstants(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

  override fun getName(): String {
    return "ALCConstants"
  }

  override fun getConstants(): MutableMap<String, Any> {
    val constants = HashMap<String, Any>()
    constants.put("NAVIGATION_EVENT", "NAVIGATION_EVENT")
    constants.put("EVENT_TYPE", "EVENT_TYPE")
    constants.put("RECLICK_TAB", "RECLICK_TAB")
    constants.put("VIEW_DID_APPEAR", "VIEW_DID_APPEAR")
    constants.put("VIEW_DID_DISAPPEAR", "VIEW_DID_DISAPPEAR")
    constants.put("COMPONENT_RESULT", "COMPONENT_RESULT")
    constants.put("RESULT_TYPE", "RESULT_TYPE")
    constants.put("RESULT_TYPE_OK", "RESULT_TYPE_OK")
    constants.put("RESULT_TYPE_CANCEL", "RESULT_TYPE_CANCEL")
    constants.put("RESULT_DATA", "RESULT_DATA")
    constants.put("SCREEN_ID", "SCREEN_ID")
    return constants
  }

}
